diff --git a/src/pprinter.cc b/src/pprinter.cc
index 2cd8178..10f6e3b 100644
--- a/src/pprinter.cc
+++ b/src/pprinter.cc
@@ -2618,7 +2618,7 @@ std::string to_string(const Material &material, const uint32_t indent, bool clos
   if (material.surface.authored()) {
 
     // TODO: list edit?.
-    ss << pprint::Indent(indent+1) << "token outputs:surface ";
+    ss << pprint::Indent(indent+1) << "token outputs:surface.connect ";
 
     const auto &conns = material.surface.get_connections();
     if (conns.size() == 1) {
@@ -2642,7 +2642,7 @@ std::string to_string(const Material &material, const uint32_t indent, bool clos
 
   if (material.displacement.authored()) {
     // TODO: list edit?.
-    ss << pprint::Indent(indent+1) << "token outputs:displacement ";
+    ss << pprint::Indent(indent+1) << "token outputs:displacement.connect ";
 
     const auto &conns = material.displacement.get_connections();
     if (conns.size() == 1) {
@@ -2666,7 +2666,7 @@ std::string to_string(const Material &material, const uint32_t indent, bool clos
 
   if (material.volume.authored()) {
     // TODO: list edit?.
-    ss << pprint::Indent(indent+1) << "token outputs:volume ";
+    ss << pprint::Indent(indent+1) << "token outputs:volume.connect ";
 
     const auto &conns = material.volume.get_connections();
     if (conns.size() == 1) {
